spec/models/component_spec.rb

```diff
@@ -16,8 +16,56 @@
 
 RSpec.describe Component, type: :model do
   describe 'should not be valid' do
+    describe 'if it has' do
+      let!(:component) { build :component }
+      let!(:invalid_component) { create :component, position: 2, subsection_id: 1}
+
+      it 'no component type' do
+        component.component_type = nil
+        expect(component.valid?).to be false
+      end
+
+      it 'no content url' do
+        component.content_url = nil
+        expect(component.valid?).to be false
+      end
+
+      it 'no position' do
+        component.position = nil
+        expect(component.valid?).to be false
+      end
+
+      it 'no subsection_id' do
+        component.subsection_id = nil
+        expect(component.valid?).to be false
+      end
+
+      it 'same position' do
+        component.subsection_id = invalid_component.subsection_id
+        component.position = invalid_component.position
+        expect(component.valid?).to be false
+      end
+    end
   end
 
   describe 'should be valid' do
+    describe 'if it has' do
+      let!(:component) do
+        build :component, component_type: 1,
```

Comment:
If you want to leave attributes the same, you don't need to include them (their default values are defined in the factory.